Create is_script.mdx (#334)
comcode-org/hackmud_wiki: docs/scripting/scripts.lib/is_script.mdx

---
title: scripts.lib().is_script()
---

A function that checks if the input is a valid script.

## Syntax

```js
#fs.scripts.lib().is_script(scriptname);
```

### Parameters

# scriptname

The string representation of a scriptname, e.g. "scripts.trust"

### Return

> Returns a string

Reviewer: ```suggestion
Returns a boolean.
```

## Example

```js
function(context, args) {
  const l = #fs.scripts.lib();
  const scriptname = "scripts.trust";

>   return l.is_script(scriptname);

Reviewer: ```suggestion
  return l.is_script(context, arg) ? "this was called with is_script or was called as a subscript" : "etc";
```

---
title: .is_script()
---

A function that checks if a script is being ran as a subscript, given context and args parameters. Useful for differentiating behavior in a script between calls on the CLI vs in subscripts

## Syntax

```js
#fs.scripts.lib().is_script(context, args);
```

### Parameters

#### context

A context object - typically the one received by your own script.

#### args

An arguments object - typically the one received by your own script. Used to make `{is_script:true}` on the CLI behave identically to a subscripted output.

### Return

Returns a boolean.

## Example

```js
function(context, args) {
  const l = #fs.scripts.lib();

>   return l.is_script(context, arg) ? "this was called with is_script or was called as a subscript" : "etc";

Reviewer: ```suggestion
  return l.is_script(context, arg) ? "this was called with is_script:true or was called as a subscript" : "neither";
```
